pkg/network/driver_interface.go

```diff
@@ -118,13 +143,26 @@ func (di *DriverInterface) Close() error {
 	if err := windows.CloseHandle(di.driverStatsHandle.handle); err != nil {
 		return errors.Wrap(err, "error closing stat file handle")
 	}
+
+	// destroy io completion port, and file
+	if err := windows.CancelIoEx(di.driverDNSHandle.handle, nil); err != nil {
+		return errors.Wrap(err, "error cancelling DNS io completion")
+	}
+	if err := windows.CloseHandle(di.driverDNSHandle.handle); err != nil {
+		return errors.Wrap(err, "error closing driver DNS handle")
+	}
+
+	for _, buf := range di.dnsReadBuffers {
+		C.free(unsafe.Pointer(buf))
```

Comment:
Should we empty the slice at this point?

Reply:
I assume you mean the `dnsReadBuffers` slice?  I'll change that. We don't *need* to but it would be good practice. Otherwise, two calls to Close() will try to double free